python/sklearnserver/sklearnserver/__main__.py

```diff
@@ -37,10 +37,13 @@
 
     except ModelMissingError:
         logger.error(
-            f"fail to locate model file for model {args.model_name} under dir {args.model_dir},"
+            f"failed to locate model file for model {args.model_name} under dir {args.model_dir},"
             f"trying loading from model repository."
         )
-
+        # Case 1: Model will be loaded from model repository automatically, if present
+        # Case 2: In the event that the model repository is empty, it's possible that this is a scenario for
+        # multi-model serving. In such a case, models are loaded dynamically using the TrainedModel.
+        # Therefore, we start the server without any preloaded models
         kserve.ModelServer(
             registered_models=SKLearnModelRepository(args.model_dir)
-        ).start([model])
```

Comment:
Is this the line which is cause NoModelReady exception for the multiple model case?

Reply:
Yeah, in mms the server starts without model. so this model's  `ready` attribute will be false as `model.load` will fail with ModelMissingError. Since it is marked `false` the the no model ready error is triggered